sql: flake on TestSessionMigration/max_size session serialization size error

In a PR with a new session variable I sometimes get this CI error:

```
 TestSessionMigration/max_size
    session_migration_test.go:103:
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/testdata/session_migration/max_size:13: SELECT encode(crdb_internal.serialize_session(), 'hex')
        expected:
        ERROR: crdb_internal.serialize_session(): serialized session size 5.0 KiB exceeds max allowed size 2.0 KiB (SQLSTATE 54000)
        found:
        ERROR: crdb_internal.serialize_session(): serialized session size 5.1 KiB exceeds max allowed size 2.0 KiB (SQLSTATE 54000)
    --- FAIL: TestSessionMigration/max_size (0.41s)
```

Jira issue: CRDB-18460

--- a/pkg/sql/session_migration_test.go
+++ b/pkg/sql/session_migration_test.go
@@ -14,6 +14,7 @@ import (
 	"context"
 	"fmt"
 	"net/url"
+	"regexp"
 	"strings"
 	"testing"
 
@@ -209,6 +210,30 @@ WHERE dump.variable IS NULL OR dump2.variable IS NULL OR dump.variable != dump2.
 					return err.Error()
 				}
 				return ret
+
+			case "error":
+				var errorRE string
+				for _, arg := range d.CmdArgs {
+					if arg.Key == "regexp" {
+						if len(arg.Vals) != 1 {
+							t.Fatalf("regexp arg expects one value")
+						}
+						errorRE = arg.Vals[0]
+					}
+				}
+				if errorRE == "" {
+					t.Fatalf("error requires regexp arg")
+				}
+				_, err := dbConn.Exec(ctx, getQuery())
+				if err != nil {
+					if ok, err := regexp.MatchString(errorRE, err.Error()); ok {
+						return ""
+					} else {
+						require.NoError(t, err)
+					}
+					t.Fatalf("error regexp didn't match: %s", err.Error())
+				}
+				t.Fatalf("expected error")
 			}
 			t.Fatalf("unknown command: %s", d.Cmd)
 			return "unexpected"

--- a/pkg/sql/testdata/session_migration/max_size
+++ b/pkg/sql/testdata/session_migration/max_size
@@ -10,7 +10,6 @@ exec
 PREPARE q AS SELECT '$large_string'
 ----
 
-exec
+error regexp=(ERROR: crdb_internal.serialize_session\(\): serialized session size [[:digit:]]\.[[:digit:]] KiB exceeds max allowed size 2\.0 KiB \(SQLSTATE 54000\))
 SELECT encode(crdb_internal.serialize_session(), 'hex')
 ----
-ERROR: crdb_internal.serialize_session(): serialized session size 5.0 KiB exceeds max allowed size 2.0 KiB (SQLSTATE 54000)

